src/Okta.Idx.Sdk/IdxClient.cs

```diff
@@ -237,6 +240,106 @@ internal async Task<IIdxResponse> IntrospectAsync(IIdxContext idxContext, Cancel
                 request, cancellationToken).ConfigureAwait(false);
         }
 
+        /// <inheritdoc/>
+        public async Task<WidgetSignInResponse> StartWidgetSignInAsync(CancellationToken cancellationToken = default)
+        {
+            IIdxContext idxContext = await this.InteractAsync();
+            return new WidgetSignInResponse
+            {
+                IdxContext = idxContext,
+                SignInWidgetConfiguration = new SignInWidgetConfiguration(this.Configuration, idxContext),
+            };
+        }
+
+        /// <inheritdoc/>
+        public async Task<OktaTokens> RedeemInteractionCodeAsync(IIdxContext idxContext, string interactionCode, Action<Exception> exceptionHandler = null, CancellationToken cancellationToken = default)
```

Comment:
We already have a `TokenResponse` object that we return when exchanging tokens. Can we pls reuse that?

Also, the current approach of the SDK is throwing an `OktaException` ( or a child of) and let the consumer catch and deal with that at their app level. I suggest removing the `exceptionHandler` to be consistent with the existing patterns.